tests/sources/test_poetry_pyproject.py

```diff
@@ -0,0 +1,289 @@
+import textwrap
+from io import StringIO
+from typing import List
+
+import pytest
+
+from check_python_versions.sources.poetry_pyproject import (
+    get_python_requires,
+    get_supported_python_versions,
+    update_python_requires,
+    update_supported_python_versions,
+)
+from check_python_versions.versions import Version
+
+
+def v(versions: List[str]) -> List[Version]:
+    return [Version.from_string(v) for v in versions]
+
+
+def test_get_supported_python_versions(tmp_path):
+    filename = tmp_path / "pyproject.toml"
+    filename.write_text(textwrap.dedent("""\
+        [tool.poetry]
+            name='foo',
+            classifiers=[
+                'Programming Language :: Python :: 2.7',
+                'Programming Language :: Python :: 3.6',
+                'Programming Language :: Python :: 3.10',
+            ]
+    """))
+    assert get_supported_python_versions(str(filename)) == v(['2.7', '3.6', '3.10'])
+
+
+def test_get_supported_python_versions_computed(tmp_path):
+    filename = tmp_path / "pyproject.toml"
+    filename.write_text(textwrap.dedent("""\
+        [tool.poetry]
+            name='foo',
+            classifiers=[
+                'Programming Language :: Python :: %s' % v
+                for v in ['2.7', '3.7']
+            ]
+    """))
+    assert get_supported_python_versions(str(filename)) == v(['2.7', '3.7'])
+
+
+def test_get_supported_python_versions_string(tmp_path, capsys):
+    filename = tmp_path / "pyproject.toml"
+    filename.write_text(textwrap.dedent("""\
+        [tool.poetry]
+            name='foo',
+            classifiers='''
+                Programming Language :: Python :: 2.7
+                Programming Language :: Python :: 3.6
+            '''
+    """))
+    assert get_supported_python_versions(str(filename)) == []
+    assert (
+        "The value passed to poetry classifiers is not a list"
+        in capsys.readouterr().err
+    )
+
+
+def test_get_supported_python_versions_from_file_object_cannot_run_pyproject_toml():
+    fp = StringIO(textwrap.dedent("""\
+        [tool.poetry]
+            name='foo',
+            classifiers=[
+                'Programming Language :: Python :: %s' % v
+                for v in ['2.7', '3.7']
```

Comment:
Is this actually valid syntax?

Reply:
> Is this actually valid syntax?

no, I'm going to remove the tests that uses this declaration way